File: javascript/core/src/engine/verbs/unorder.ts
/*!
 * Copyright (c) Microsoft. All rights reserved.
 * Licensed under the MIT license. See LICENSE file in the project.
 */
import { makeStepFunction, makeStepNode, wrapColumnStep } from '../factories.js'

const doUnorder = wrapColumnStep(input => input.unorder())

export const unorder = makeStepFunction(doUnorder)
export const unorderNode = makeStepNode(doUnorder)

File: python/datashaper/datashaper/verbs/unorder.py
#
# Copyright (c) Microsoft. All rights reserved.
# Licensed under the MIT license. See LICENSE file in the project.
#
"""Unorder verb implementation."""

from typing import Any

import pandas as pd

from .decorators import OutputMode, inputs, verb, wrap_verb_result


@verb(
    name="unorder",
    immutable_input=True,
    adapters=[
        inputs(default_input_argname="table"),
        wrap_verb_result(mode=OutputMode.Table),
    ],
)
def unorder(table: pd.DataFrame, **_kwargs: Any) -> pd.DataFrame:
    """Unorder verb implementation."""
    return table.sort_index()
